jwst/ramp_fitting/ramp_fit.py

```diff
@@ -1261,94 +1333,83 @@ def ramp_fit_second_pass(
     if segs_4 is not None:
         del segs_4
 
+    input_model.data = data
+    input_model.err = err
+    input_model.groupdq = groupdq
+    input_model.pixeldq = inpixeldq
+
     return var_p3, var_r3, var_p4, var_r4, var_both4, var_both3, inv_var_both4, \
         s_inv_var_p3, s_inv_var_r3, s_inv_var_both3
 
 
-def ramp_fit_overall_slopes_and_variances(
-        n_int, nrows, ncols, imshape, orig_cubeshape, orig_nreads, buffsize,
-        groupdq, first_pass_ans, second_pass_ans, opt_res, save_opt,
-        int_times, instrume, tstart, nframes, nreads, groupgap, dropframes1):
+def ramp_fit_overall(
+        input_model, orig_cubeshape, orig_ngroups, buffsize, fit_slopes_ans,
+        variances_ans, save_opt, int_times, tstart):
     """
-
     Parameter
     ---------
-    n_int : int
-        Number of integrations
-
-    nrows : int
-        Number of rows in image
-
-    ncols : int
-        Number of columns in image
-
-    imshape : (int, int)
-       Shape of image
+    input_model : data model
+        input data model, assumed to be of type RampModel
 
     orig_cubeshape : (int, int, int) tuple
        Original shape of input dataset
 
-    orig_nreads : int
+    orig_ngroups: int
        Original number of reads
```

Comment:
Should modify this comment here too, to change "reads" to "groups"

Reply:
Thank you.  Since our discussion before I have been removing references to "read" that really indicates "group" in JWST.  I changed this documentation.